crypto/evp/evp_fetch.c

```diff
@@ -334,7 +334,7 @@ inner_evp_generic_fetch(struct evp_method_data_st *methdata,
                 method = NULL;
             } else {
                 meth_id = evp_method_id(name_id, operation_id);
-                if (name_id != 0)
+                if (meth_id != 0)
```

Comment:
Doesnt seem like a big issue either way since a value of 0 here would indicate either the name_id or operation_id were out of range (which is highly unlikely).